### PR TITLE
managed data of typing and added delete btn to delete user name

### PR DESCRIPTION
**Pull request by weberankit**
managed data of typing and added a delete button to delete the user name
features added
1-delete button provided to delete userName
2-now the user can delete the history of typing data without changing userName(_means no need to provide username again and again after deleting user typing history).
Note--if the user wants to delete all data then the user has to click on the button reset and delete

fixed--
chrome_6jenpjcuPQ
video--
you can skip video from 0.31 to 1:43 (this time taken to pass the game)
https://user-images.githubusercontent.com/94105514/220374784-475cb16b-03a7-4063-be93-46ecf2d1bed0.mp4

### CODE REVIEW

**Review by arpitghura — COMMENTED** [review]
<h1>It's great having you contribute to this project</h1> Welcome to the community :nerd_face: Thanks for giving your time to contribute to <b>Typing Test</b> Project. <br/> You should join our <a href='https://discord.gg/7c3VGDutmS'>Discord</a> chat - we help and encourage each other to contribute to open source little and often

**Comment by weberankit** [comment]
solved issue #115

**Comment by arpitghura** [comment]
Style the `delete username` button accordingly.

**Comment by weberankit** [comment]
plz review it 
delete button style is done and plus hide delete button when there is no username avail in the local storage
hide button
Screenshot (14998)
style button
chrome_0gDHZsnibm
added hover effect also
chrome_6jenpjcuPQ

**Comment by arpitghura** [comment]
> plz review it delete button style is done and plus hide delete button when there is no username avail in the local storage hide button Screenshot (14998) style button chrome_0gDHZsnibm added hover effect also Screenshot (15000)

Do styling similar to the `reset data` button but change the background also the current background has readability issues. Could you try some other variants of the color?

**Comment by weberankit** [comment]
plz check 
chrome_pDweDpxCIY

File: assets/js/profile.js [hunk 1]
```diff
@@ -3,13 +3,19 @@ const userNameEle = document.getElementById("userName");
...
         linktoprofile.innerText = "Click here";
         linktoprofile.href = "./practice.html";
-        navigateEle.innerText = " to set up your profile";
+        navigateEle.innerText = `${msg}`;
+        navigateEle.insertAdjacentElement('afterbegin', linktoprofile);
+    }else{
+        const linktoprofile = document.createElement('a');
+        linktoprofile.innerText = "Click here";
+        linktoprofile.href = "./practice.html";
+        navigateEle.innerText = "to play again";
```
Comment (arpitghura) [inline assets/js/profile.js:18]: Change `to play again` to `to practice again`.

File: assets/js/profile.js [hunk 2]
```diff
@@ -105,8 +112,31 @@ showHistory();
...
+   //to show msg after delete data
+   checkIfUserData("to play again")
+})
+
+//after click on delete username button 
+const deleteUserName=document.querySelector(".reset-userName")
+  deleteUserName.addEventListener("click",(e)=>{
+    localStorage.removeItem("typerName");
+    e.target.style.display="none"
+    checkIfUserData(" to set up your name and for play again")
```
Comment (arpitghura) [inline assets/js/profile.js:128]:
```suggestion
    checkIfUserData(" to set up your name and practice again")
```

File: assets/js/profile.js [hunk 3]
```diff
@@ -105,8 +112,31 @@ showHistory();
...
+    getUserName()
 })
+
+
+//hide username 
+const usernamefromLocal = localStorage.getItem("typerName");
+if(usernamefromLocal==null){
+    deleteUserName.style.display="none"
+    checkIfUserData(" to set up your name and for play again")
```
Comment (arpitghura) [inline assets/js/profile.js:137]:
```suggestion
    checkIfUserData(" to set up your name and practice again")
```

File: profile.html [hunk 4]
```diff
@@ -65,6 +65,7 @@
             <div class="container">
                 <h1 class="heading">Profile Details</h1>
                 <h2 class="subtitle" id="userName"></h2>
+                <button class="reset-userName">delete Username</button>
```
Comment (arpitghura) [inline profile.html:68]:
```suggestion
                <button class="reset-userName">Reset Username</button>
```

**Review by arpitghura — CHANGES_REQUESTED** [review]
Do the required changes

File: assets/js/profile.js (hunk 1 above)
Comment (weberankit) [reply assets/js/profile.js:18]: ok

File: assets/js/profile.js (hunk 3 above)
Comment (weberankit) [reply assets/js/profile.js:137]: ok

File: assets/js/profile.js (hunk 2 above)
Comment (weberankit) [reply assets/js/profile.js:128]: ok

**Comment by weberankit** [comment]
plz check it

File: profile.html (hunk 4 above)
Comment (weberankit) [reply profile.html]: Ok

File: assets/js/profile.js [hunk 5]
```diff
@@ -105,8 +112,31 @@ showHistory();
...
+   //to show msg after delete data
+   checkIfUserData(" to practice again")
+})
+
+//after click on delete username button 
+const deleteUserName=document.querySelector(".reset-userName")
+  deleteUserName.addEventListener("click",(e)=>{
+    localStorage.removeItem("typerName");
+    e.target.style.display="none"
+    checkIfUserData(" to set up your name and practice again")
```
Comment (arpitghura) [inline assets/js/profile.js:128]: if the user visits the website for the first time. The text appears will ' to set up your name and practice again '. the word again didn't make any sense as the user visits for the first time. so remove it.

**Review by arpitghura — CHANGES_REQUESTED** [review]

File: assets/js/profile.js (hunk 5 above)
Comment (weberankit) [reply assets/js/profile.js]: done

**Comment by weberankit** [comment]
image
plz check it

**Review by arpitghura — APPROVED** [review]
Looks Good

**Pull request merged**